src/memray/_memray.pyx

```diff
@@ -199,37 +208,66 @@ cdef class AllocationRecord:
 
     def hybrid_stack_trace(self, max_stacks=None):
         cdef vector[unsigned char] is_entry_frame
-        cdef list python_stack
-        if max_stacks is None:
-            python_stack = self._reader.get().Py_GetStackFrameAndEntryInfo(
-                self._tuple[4], &is_entry_frame
-            )
-        else:
-            python_stack = self._reader.get().Py_GetStackFrameAndEntryInfo(
-                self._tuple[4], &is_entry_frame, max_stacks
-            )
 
-        native_stack = self.native_stack_trace(max_stacks)
+        python_stack = self._reader.get().Py_GetStackFrameAndEntryInfo(
+            self._tuple[4], &is_entry_frame
+        )
+        native_stack = self.native_stack_trace()
 
-        if not python_stack:
-            yield from native_stack
-            return
+        cdef ssize_t num_non_entry_frames = 0
+        for val in is_entry_frame:
+            if not val:
+                num_non_entry_frames += 1
```

Comment:
```suggestion
        cdef ssize_t num_non_entry_frames = sum(int(not val) for val in is_entry_frame)
```

Same concern as before regarding going back and forth to python and C

Reply:
This generates some pretty not-great code - not just a conversion to Python, but a whole generator expression closure.

Really, since `is_entry_frame` is a vector, what we want is just `std::count(is_entry_frame.begin(), is_entry_frame.end(), 0)` - but Cython doesn't include a declaration for `std::count`, and I was being lazy. I'll just add one - it's more code, but more elegant.